### PR TITLE
Requirement API v1.0

### PR DESCRIPTION
**Pull request by handotdev**
### Summary

This pull request is the first step towards implementing a full-service requirements backend

- [x] Add the first version of the requirements API (v1.0) with basic query functionalities
- [x] Developed the mongo database structure with logic provided in the docs
- [x] Add University, Engineering, and CAS requirements to the database
- [x] Add supporting documentation to understanding the API and the data structure

### Test Plan 

[Screen-Shot-2019-10-24-at-10-59-01-PM.png](https://postimg.cc/BPs72MD7)
As of 11:00 PM October 4, request status, message, and metadata needs to be added to request result

### CODE REVIEW

File: requirements/reqs-data.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
+exports.requirements = {
```
Comment (SamChou19815) [inline requirements/reqs-data.js]: Make this file json instead?

File: api/api.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const request = require('request');
```
Comment (SamChou19815) [inline api/api.js:1]: Let's fix all the linter errors and warnings first. You can go to `Files changed` tab on github or run `npm run lint` locally.

File: api/api.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                            return callback(result);
+                        }
+                    })
+                }
+            });
+        });
+    });
+}
+
+function getSubjects(ros, callback) {
```
Comment (SamChou19815) [inline api/api.js]: Using `async/await` may make the code more readable.
Also, consider adding jsdoc to the parameters to make document its usage, especially for the function that involves callbacks. Or even better, use typescript :)

**Review by SamChou19815 — CHANGES_REQUESTED** [review]

File: api/api.js (hunk 2 above)
Comment (handotdev) [reply api/api.js:1]: On it

File: requirements/reqs-data.js (hunk 1 above)
Comment (handotdev) [reply requirements/reqs-data.js]: Good idea. Fixed that. Gonne commit soon

**Review by SamChou19815 — COMMENTED** [review]
Is `api/api.js` used anywhere?

**Comment by handotdev** [comment]
Not at the moment, but it will be used to scrape, parse, and store all of Cornell's course data

**Review by SamChou19815 — APPROVED** [review]
@hanywang2 I'm giving it a pass right now, but in the future it would be better if every diff is self-contained and only does one thing at a time.

**Pull request merged**